### PR TITLE
calibre: 6.17.0 -> 6.19.1

### PR DESCRIPTION
**Pull request by kilianar**
###### Description of changes
- https://github.com/kovidgoyal/calibre/releases/tag/v6.18.0
- https://github.com/kovidgoyal/calibre/releases/tag/v6.18.1
- https://github.com/kovidgoyal/calibre/releases/tag/v6.19.0
- https://github.com/kovidgoyal/calibre/releases/tag/v6.19.1

A detailed description of the changes can be found in https://calibre-ebook.com/whats-new.

Updating Calibre to the latest release necessitates podofo version 0.10.0, given its API/ABI incompatibility with versions 0.9.x. To accommodate this disparity, I included the newer version as an individual package instead of creating a generic package, as the source, dependencies, and make flags have also undergone changes.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/podofo/0.10.x.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  postInstall = ''
+    moveToOutput lib "$lib"
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/podofo/podofo";
+    description = "A library to work with the PDF file format";
+    platforms = platforms.all;
+    license = with licenses; [ gpl2Plus lgpl2Plus ];
+  };
```
Comment (AndersonTorres) [inline pkgs/development/libraries/podofo/0.10.x.nix:65]:
```suggestion
  meta = {
    homepage = "https://github.com/podofo/podofo";
    description = "A library to work with the PDF file format";
    platforms = lib.platforms.all;
    license = with lib.licenses; [ gpl2Plus lgpl2Plus ];
    maintainers = [];
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/podofo/0.10.x.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    libpng
+    libtiff
+    libxml2
+    lua5
+    openssl
+    zlib
+  ];
+
+  cmakeFlags = [
+    "-DPODOFO_BUILD_STATIC=OFF"
```
[the remark this answers is not in the document]
Comment (kilianar) [reply pkgs/development/libraries/podofo/0.10.x.nix]: Thank you for catching this issue. I suppose that change should be made in the [original package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/podofo/default.nix) as well.

**Review by AndersonTorres — APPROVED** [review]
Ping me in 4 days and I will merge.

**Comment by pbek** [comment]
Result of `nixpkgs-review pr 235002` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>calibre</li>
    <li>podofo010</li>
    <li>podofo010.dev</li>
    <li>podofo010.lib</li>
  </ul>
</details>

**Review by pbek — APPROVED** [review]

**Comment by pbek** [comment]
Calibre seems to run nice for me.

**Comment by risicle** [comment]
#230772

**Pull request merged**